### DISCUSSION TITLE
Decrease monitor time onedrive as a service

### DISCUSSION THREAD

**Discussion by jackal1-66** [abraunegg/onedrive#1893] [Q&A] [answered]
Hello, I've been using this amazing program for a while now, but I was wondering if there was a way to decrease the time of the syncing cycle when Onedrive is started as a service. Right now each cycle happens every 20 minutes, which is quite a lot... 
If it was realtime when editing remote files it would be wonderful, but even 1 minute wouldn't be bad.

**Comment by abraunegg** [comment, marked as answer]
@jackal1-66 

> Hello, I've been using this amazing program for a while now, but I was wondering if there was a way to decrease the time of the syncing cycle when Onedrive is started as a service. Right now each cycle happens every 20 minutes, which is quite a lot... 

It is odd that your default is 20 mins - as, the application default is 300 seconds (5 mins).

Look at your application configuration via `onedrive --display-config` - be mindful here that if your service configuration is using a specific configuration file you need to also pass in the `--confdir <path>` to ensure you are displaying the right configuration - example:
```
[alex@centos7full onedrive]$ ./onedrive --display-config
Configuration file successfully loaded
onedrive version                       = v2.4.16-1-g8271adc
Config path                            = /home/alex/.config/onedrive
Config file found in config path       = true
Config option 'check_nosync'           = false
Config option 'sync_dir'               = /home/alex/OneDrive
Config option 'skip_dir'               = 
Config option 'skip_file'              = ~*|.~*|*.tmp
Config option 'skip_dotfiles'          = false
Config option 'skip_symlinks'          = false
Config option 'monitor_interval'       = 300
Config option 'min_notify_changes'     = 5
Config option 'log_dir'                = /var/log/onedrive/
Config option 'classify_as_big_delete' = 1000
Config option 'upload_only'            = false
Config option 'no_remote_delete'       = false
Config option 'remove_source_files'    = false
Config option 'sync_root_files'        = false
Selective sync 'sync_list' configured  = false
Business Shared Folders configured     = false
[alex@centos7full onedrive]$ 
```
```
[alex@centos7full onedrive]$ ./onedrive --confdir '~/.config/onedrive-personal/' --display-config
Configuration file successfully loaded
onedrive version                       = v2.4.16-1-g8271adc
Config path                            = /home/alex/.config/onedrive-personal/
Config file found in config path       = true
Config option 'check_nosync'           = false
Config option 'sync_dir'               = /home/alex/OneDrivePersonal
Config option 'skip_dir'               = 
Config option 'skip_file'              = ~*|.~*|*.tmp
Config option 'skip_dotfiles'          = false
Config option 'skip_symlinks'          = false
Config option 'monitor_interval'       = 60
Config option 'min_notify_changes'     = 1
Config option 'log_dir'                = /var/log/onedrive/
Config option 'classify_as_big_delete' = 1000
Config option 'upload_only'            = false
Config option 'no_remote_delete'       = false
Config option 'remove_source_files'    = false
Config option 'sync_root_files'        = false
Selective sync 'sync_list' configured  = false
Business Shared Folders configured     = false
```

The option you are looking for is 'monitor_interval' - edit your applicable 'config' file and then retest your configuration, then restart your services as required - otherwise the change will not be picked up.

My advice is do not go too low - otherwise, the client will forever be running - consuming CPU & power and if on a laptop, draining your battery. 300 seconds (5 mins) is the right value here.

> If it was realtime when editing remote files it would be wonderful, but even 1 minute wouldn't be bad.

You can get realtime downloads from OneDrive by using 'webhooks' - but you need to configure this - refer to https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#use-webhook-to-subscribe-to-remote-updates-in-monitor-mode

**Reply by jackal1-66** [reply]
Thank you very much for the answer, the configuration file indeed shows 300s and apparently now it's working fine.
I think the issue was just a buggy session of Ubuntu which didn't update well enough the service. 
Regarding the webhook though I've got a question: Considering that I am on a notebook and I change continuosly my 
ip address due to different connections, is there a way to account for that when creating the webhook or do I have to 
change the value manually every single time?

**Reply by abraunegg** [reply]
@jackal1-66 
> Thank you very much for the answer, the configuration file indeed shows 300s and apparently now it's working fine. I think the issue was just a buggy session of Ubuntu which didn't update well enough the service.

Just make sure you are using v2.4.16 and the packages from the OpenSuSE Build Service. Make sure you read: https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md

> Regarding the webhook though I've got a question: Considering that I am on a notebook and I change continuosly my
ip address due to different connections, is there a way to account for that when creating the webhook or do I have to
change the value manually every single time?

You will need to research and work on a solution that works for you - an external URL is what you need, then how that URL knows your current IP address and gets automatically updated .. all of this is out of scope for this discussion.
